CMakeLists.txt

```diff
@@ -63,6 +63,28 @@ option(ENABLE_ROCSOLVER_BACKEND "Enable the rocSOLVER backend for the LAPACK int
 # dft
 option(ENABLE_CUFFT_BACKEND "Enable the cuFFT backend for the DFT interface" OFF)
 option(ENABLE_ROCFFT_BACKEND "Enable the rocFFT backend for the DFT interface" OFF)
+option(ENABLE_PORTFFT_BACKEND_INTEL_CPU "Enable the portFFT DFT backend for CPU" OFF)
+option(ENABLE_PORTFFT_BACKEND_INTEL_GPU "Enable the portFFT DFT backend for Intel GPU" OFF)
+option(ENABLE_PORTFFT_BACKEND_AMD_GPU "Enable the portFFT DFT backend for AMD GPU" OFF)
+option(ENABLE_PORTFFT_BACKEND_NVIDIA_GPU "Enable the portFFT DFT backend for NVIDIA GPU" OFF)
+
+if (ENABLE_PORTFFT_BACKEND_INTEL_CPU AND ENABLE_MKLCPU_BACKEND)
+  message(SEND_ERROR "ENABLE_PORTFFT_BACKEND_INTEL_CPU and ENABLE_MKLCPU_BACKEND cannot be enabled at the same time. \
+  Only one DFT backend can be enable for each device type.")
+endif()
+if (ENABLE_PORTFFT_BACKEND_INTEL_GPU AND ENABLE_MKLGPU_BACKEND)
+  message(SEND_ERROR "ENABLE_PORTFFT_BACKEND_INTEL_GPU and ENABLE_MKLGPU_BACKEND cannot be enabled at the same time. \
+  Only one DFT backend can be enable for each device type.")
+endif()
+if (ENABLE_PORTFFT_BACKEND_AMD_GPU AND ENABLE_ROCFFT_BACKEND)
+  message(SEND_ERROR "ENABLE_PORTFFT_BACKEND_AMD_GPU and ENABLE_ROCFFT_BACKEND cannot be enabled at the same time. \
+  Only one DFT backend can be enable for each device type.")
+endif()
+if (ENABLE_PORTFFT_BACKEND_NVIDIA_GPU AND ENABLE_CUFFT_BACKEND)
+  message(SEND_ERROR "ENABLE_PORTFFT_BACKEND_NVIDIA_GPU and ENABLE_CUFFT_BACKEND cannot be enabled at the same time. \
+  Only one DFT backend can be enable for each device type.")
+endif()
+set(ENABLE_PORTFFT_BACKEND (ENABLE_PORTFFT_BACKEND_AMD_GPU || ENABLE_PORTFFT_BACKEND_INTEL_CPU ||ENABLE_PORTFFT_BACKEND_INTEL_GPU ||ENABLE_PORTFFT_BACKEND_NVIDIA_GPU))
```

Comment:
We don't need all of this logic for portFFT. I think it would be simpler if we only have `ENABLE_PORTFFT_BACKEND` (and same comment for the compiler defines).
I suppose this could let users combine portfft with other backends for different devices but the original idea was that portFFT should be enabled for all or nothing since the whole point is that it is portable. For portBLAS we check that the portBLAS backend cannot be used with other backends [here](https://github.com/oneapi-src/oneMKL/blob/develop/CMakeLists.txt#L118).

Reply:
Addressed in https://github.com/FMarno/oneMKL/pull/1/commits/57e39ab53bb2a6d15dd4a9592f11517907953d53